Add article for CSSAnimation (#11)
GoogleChromeLabs/puppy-content: content/en-US/api/CSSAnimation/CSSAnimation.md

---
recipe: api-feature
title: 'CSSAnimation'
mdn_url: /en-US/docs/Web/API/CSSAnimation
specifications: https://drafts.csswg.org/css-animations-2/#the-CSSAnimation-interface
browser_compatibility: api.CSSAnimation

---

## Description

> The `CSSAnimation` interface of the [CSS Animations Level
> 2](https://drafts.csswg.org/css-animations-2/) represents a CSS animation as an
> [`Animation`](/en-US/docs/Web/API/Animation/Animation) object in the
> [Web Animations API](/en-US/docs/Web/API/Web_Animations_API) model.
> 
> As such it extends the [`Animation`](/en-US/docs/Web/API/Animation/Animation)
> interface and provide playback controls and the ability to change various
> aspects of the CSS animation.

Reviewer: ```suggestion
The `CSSAnimation` interface of the [Web Animations API](/en-US/docs/Web/API/Web_Animations_API) provides playback controls and the ability to change various
aspects of the CSS animation.
```

Reviewer: Inheritance and spec links are handled in a different (and standard) way on MDN. The spec section is not in the boilerplate because I build it from the metadata in a script. You can [see the result here](https://developer.mozilla.org/en-US/docs/Web/API/AbsoluteOrientationSensor#Specifications).

Author: Just a note that CSSAnimation interface is defined in CSS Animations Level-2 and extends the Animation interface that is defined Web Animations API. So I am not sure exactly "interface the [spec]" means in this context.

Just wanted to mention this but will leave its interpretation up to you. :)

Reviewer: Your intro describes what CSSAnimation inherits from and provides a link to the spec. I simply meant that both bits of information are handled by different but standard ways.

> 
> 
> ## Constructor
> 
> None.
> 
> It is not possible to construct a `CSSAnimation` in JavaScript. Instead instance
> of this class may be obtained via
> [`Element.getAnimations()`](/en-US/docs/Web/API/Element/getAnimations) or
> [`Document.getAnimations()`](/en-US/docs/Web/API/Document/getAnimations).

Reviewer: ```suggestion
It is not possible to construct a `CSSAnimation` in JavaScript. Instead instance
of this class may be obtained via
[`Element.getAnimations()`](/en-US/docs/Web/API/Element/getAnimations) or
[`Document.getAnimations()`](/en-US/docs/Web/API/Document/getAnimations).
```

> ## Properties

Reviewer: ```suggestion
## Properties

__Inherits properties from its ancestor, {{domxref('Animation')}}.__
```

**`CSSAnimation.animationName`**

Returns the value of the `'animation-name'` property that created the CSS
animation and caused this object to be generated.

> 
> 

Reviewer: ```suggestion

## Methods

__No specific methods; inherits methods from its ancestor, {{domxref('Animation')}}.__
```
